core/trino-main/src/test/java/io/trino/sql/planner/assertions/JoinMatcher.java

```diff
@@ -43,11 +45,14 @@
 import static io.trino.sql.planner.ExpressionExtractor.extractExpressions;
 import static io.trino.sql.planner.assertions.MatchResult.NO_MATCH;
 import static io.trino.sql.planner.assertions.PlanMatchPattern.DynamicFilterPattern;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.equiJoinClause;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.node;
 import static io.trino.sql.planner.optimizations.PlanNodeSearcher.searchFrom;
+import static io.trino.sql.tree.ComparisonExpression.Operator.EQUAL;
 import static io.trino.sql.tree.ComparisonExpression.Operator.IS_DISTINCT_FROM;
 import static java.util.Objects.requireNonNull;
 
-final class JoinMatcher
+public final class JoinMatcher
```

Comment:
is it needed?

Reply:
Yes, this is needed to define consumers/lambdas outside package. Without `public` you will receive `JoinMatcher.Builder is defined in an inaccessible class or interface`.